src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/TintColor/IconTintColorBehavior.android.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Android.Graphics;
+using Android.Widget;
+using Microsoft.Maui.Platform;
+using AButton = Android.Widget.Button;
+using Color = Microsoft.Maui.Graphics.Color;
+using AView = Android.Views.View;
+using System.ComponentModel;
+using ImageButton = Microsoft.Maui.Controls.ImageButton;
+
+namespace CommunityToolkit.Maui.Behaviors;
+
+public partial class IconTintColorBehavior
+{
+	/// <inheritdoc/>
+	protected override void OnAttachedTo(View bindable, AView platformView) =>
+		ApplyTintColor(bindable, platformView);
+
+	/// <inheritdoc/>
+	protected override void OnDetachedFrom(View bindable, AView platformView) =>
+		ClearTintColor(bindable, platformView);
+
+	void ApplyTintColor(View element, AView control)
+	{
+		var color = TintColor;
+		element.PropertyChanged += OnElementPropertyChanged;
+
+		switch (control)
+		{
+			case ImageView image:
+				SetImageViewTintColor(image, color);
+				break;
+			case AButton button:
+				SetButtonTintColor(button, color);
+				break;
+			default:
+				throw new InvalidOperationException($"The control must be of type {nameof(AButton)} or {nameof(ImageView)}.");
+		}
+
+
+		static void SetImageViewTintColor(ImageView image, Color? color)
+		{
+			if (color is null)
+			{
+				image.ClearColorFilter();
+				color = Colors.Transparent;
+			}
+
+			image.SetColorFilter(new PorterDuffColorFilter(color.ToPlatform(), PorterDuff.Mode.SrcIn ?? throw new NullReferenceException()));
```

Comment:
Let's add some explanation to this exception to help us and fellow devs more easily find it in the source code if/when it is thrown

```suggestion
			image.SetColorFilter(new PorterDuffColorFilter(color.ToPlatform(), PorterDuff.Mode.SrcIn ?? throw new InvalidOperationException("PorterDuff.Mode.SrcIn should not be null at runtime")));
```